templates/_helpers.tpl

```diff
@@ -250,13 +242,41 @@ Return concourse environment variables for worker configuration
 - name: CONCOURSE_GARDEN_REQUEST_TIMEOUT
   value: {{ .Values.concourse.worker.garden.requestTimeout | quote }}
 {{- end }}
-{{- if .Values.concourse.worker.garden.dnsServer }}
-- name: CONCOURSE_GARDEN_DNS_SERVER
```

Comment:
Hey @muntac! Does it mean dns server won't be configurable if using guardian (I mean, using flags)? Or the idea is to use the config file for Guardian config?

Reply:
Hi! Good question. This part is a bit confusing so it will need a slightly long explanation.

There are two types of CONCOURSE_GARDEN_ flags. 

The first type is defined in the concourse binary and is used by concourse to do specific things. These are `CONCOURSE_GARDEN_BIN`, `CONCOURSE_GARDEN_CONFIG` and `CONCOURSE_GARDEN_REQUEST_TIMEOUT`.

The second type are flags that are features of the Guardian binary. For these concourse just grabs the flag, e.g. `CONCOURSE_GARDEN_DNS_SERVER`, and passes `--dns-server` to the `gdn server` command of the Guardian binary.

In Helm charts we don't define any of the second types of flags, and users can pass them I believe by putting them in the `values.yml` or in the config file like you mentioned. The reason `CONCOURSE_GARDEN_DNS_SERVER` was here before was because we had started defining it in the concourse binary to trigger a behaviour in containerd. But that behaviour in containerd is now triggered by `CONCOURSE_CONTAINERD_DNS_SERVER` so this is no longer needed.

Reply:
Thanks for clarifying ! :)